packages/core-flows/src/order/steps/set-tax-lines-for-items.ts

```diff
@@ -0,0 +1,128 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  CreateLineItemTaxLineDTO,
+  CreateShippingMethodTaxLineDTO,
+  IOrderModuleService,
+  ItemTaxLineDTO,
+  OrderDTO,
+  ShippingTaxLineDTO,
+} from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  order: OrderDTO
+  item_tax_lines: ItemTaxLineDTO[]
+  shipping_tax_lines: ShippingTaxLineDTO[]
+}
+
+export const setOrderTaxLinesForItemsStepId = "set-order-tax-lines-for-items"
+export const setOrderTaxLinesForItemsStep = createStep(
+  setOrderTaxLinesForItemsStepId,
+  async (data: StepInput, { container }) => {
+    const { order, item_tax_lines, shipping_tax_lines } = data
+    const orderService = container.resolve<IOrderModuleService>(
+      ModuleRegistrationName.ORDER
+    )
+
+    const getShippingTaxLinesPromise =
+      await orderService.listShippingMethodTaxLines({
+        shipping_method_id: shipping_tax_lines.map((t) => t.shipping_line_id),
+      })
+
+    const getItemTaxLinesPromise = await orderService.listLineItemTaxLines({
+      item_id: item_tax_lines.map((t) => t.line_item_id),
+    })
+
+    const itemsTaxLinesData = normalizeItemTaxLinesForOrder(item_tax_lines)
+    const setItemTaxLinesPromise = itemsTaxLinesData.length
+      ? orderService.setLineItemTaxLines(order.id, itemsTaxLinesData)
+      : 0
```

Comment:
suggestion: shouldn't it be `Promise.resolve([])`?

Reply:
changed it to void 0.
there is no need to be a Promise.